### PR TITLE
[SYCL] properly report supported device partitioning in Level-Zero

### PR DESCRIPTION
**Pull request by smaslov-intel**
Make adhere to SYCL spec:

1) [info::device::partition_properties]: returns the partition properties
supported by this SYCL device; a vector of info::partition_property. If this
SYCL device cannot be partitioned into at least two sub devices then the
returned vector **must be empty**.

2) [create_sub_devices()]: If the SYCL device
does not support info::partition_property::partition_by_affinity_domain or the
SYCL device does not support the info::partition_affinity_domain provided, an
exception with the **feature_not_supported error code must be thrown**.

Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: sycl/test/on-device/basic_tests/partition_supported.cpp
Comment (romanovvlad) [inline sycl/test/on-device/basic_tests/partition_supported.cpp:54]: I suspect this test passes even without the change in the L0 plugin.
If so, I suggest writing two tests: one which makes sure that correct PI calls are done(if it doesn't exist already) and second which checks that results of PI API calls are aligned with direct calls to L0 API( see https://github.com/intel/llvm/blob/sycl/sycl/unittests/pi/cuda/test_queue.cpp#L145 )

Comment (smaslov-intel) [reply sycl/test/on-device/basic_tests/partition_supported.cpp:54]: > I suspect this test passes even without the change in the L0 plugin.

No, it does not pass without the plugin fix. Indeed, Level-Zero plugin returned that the partitioning was always supported, so the subsequent call create_sub_devices() was returning "0" devices, and the check for "<2" fired.

Comment (romanovvlad) [reply sycl/test/on-device/basic_tests/partition_supported.cpp:54]: Ok.

File: sycl/test/on-device/basic_tests/partition_supported.cpp
Comment (againull) [inline sycl/test/on-device/basic_tests/partition_supported.cpp:70]: I have only one comment regarding the test. Currently in the places where test is expected to fail error message is printed to std::cerr but program still exits successfully with code 0. Can we return non-zero code in such cases to make test fail with the LIT? As far as I understand otherwise test will always pass even though error message is printed (output is not checked with FileCheck),

**Review by againull — CHANGES_REQUESTED** [review]

File: sycl/test/on-device/basic_tests/partition_supported.cpp
Comment (smaslov-intel) [reply sycl/test/on-device/basic_tests/partition_supported.cpp:70]: Updated.

File: sycl/test/on-device/basic_tests/partition_supported.cpp
Comment (againull) [inline sycl/test/on-device/basic_tests/partition_supported.cpp:84]: Can you please add "return -1" here too?

Comment (smaslov-intel) [reply sycl/test/on-device/basic_tests/partition_supported.cpp]: OK

**Review by againull — APPROVED** [review]

**Comment by romanovvlad** [comment]
run_on_host_intel2 is a known issue which is being fixed in https://github.com/intel/llvm/pull/2766

**Comment by romanovvlad** [comment]
@smaslov-intel Can we wait until #2766 is merged or we need this ASAP?

**Comment by smaslov-intel** [comment]
> @smaslov-intel Can we wait until #2766 is merged or we need this ASAP?

It can wait.

**Comment by bader** [comment]
I think #2766 can be merged right away to unblock pre-commit testing of other PRs.

**Comment by againull** [comment]
#2766 merged, can you please rebase this PR.

**Comment by bader** [comment]
> #2766 merged, can you please rebase this PR.

Or just re-run sycl-win-x64-pr job.

**Pull request merged**
